### PR TITLE
Test 2 branch

### PR DESCRIPTION
**Pull request by mani1381**
Added test coverage for chapters 10, 11, 12, 14, and 15.

Corrected typos and improved spelling throughout chapters 8 to 15.

Enhanced cleanup procedures after executing R scripts.

Fixed the issue with cross-references duplication.

Added library imports at the beginning of the exercises.

Removed unused CSV file from the Chapter 12 exercises folder

Refactored Chapter 13 exercises to utilize downloaded file from Shiny app

### CODE REVIEW

**Comment by david-yz-liu** [comment]
@mani1381 there are a bunch of merge conflicts here. This is likely due to you creating a branch off of your previous branch, rather than the `main` branch (after I merged in your first pull request).

You should first pull from the `main` branch and then manually fix the merge conflicts (and then commit and push your updates).

File: src/03-data-analysis-in-r/03-tidy.Rmd [hunk 1]
```diff
@@ -96,7 +96,7 @@ Again, let's break down what we did with the `separate` function:
 2. `into = c(...)` specifies what columns we're separating our name into. 
 3. `sep = "_"1` specifies that each element is separated by an underscore (`_`); you can use `sep = " "` if they were separated by spaces. 
 4. `remove = TRUE` removes the original sample column, no need for duplication; setting this to `FALSE` would keep the original column. 
-5. `convert = TRUE` converts the new columns to the appropriate data format. In the original column ,the replicate number is a character value because it's part of a string, `convert` ensures that it'll be converted to a numerical value. 
+5. `convert = TRUE` converts the new columns to the appropriate data format. In the original column , the replicate number is a character value because it's part of a string, `convert` ensures that it'll be converted to a numerical value.
```
Comment (david-yz-liu) [inline src/03-data-analysis-in-r/03-tidy.Rmd:99]: Delete the space before the `,`

File: exercises/chapter 10/chapter10.Rmd [hunk 2]
```diff
@@ -15,6 +15,16 @@ Before diving into the exercises, follow the instructions below to download a 20
...
 
 
+## Load Necessary Libraries
+
+Before starting the exercises, ensure you have the necessary libraries loaded:
+
+```{r}
+# Load necessary libraries
+library(tidyverse)
+
```
Comment (david-yz-liu) [inline exercises/chapter 10/chapter10.Rmd:25]: Delete the unnecessary blank line here, and in all similar cells across the exercises

File: exercises/chapter 10/chapter10_tests.R [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    # # Run all chunks in the Rmd file to execute the exercises
+    suppressWarnings(run_all_chunks(rmd_file_path))
+    expect_equal(nrow(limited_data), 10)
+  })
+  
+  # remove the temp directory
+  unlink(tmp_dir, recursive = TRUE)
+
+})
+
```
Comment (david-yz-liu) [inline exercises/chapter 10/chapter10_tests.R:86]: There are extraneous blank lines at the end of this file, and in the other test files; delete them. Every file should end with exactly one blank line.

Comment (mani1381) [reply exercises/chapter 10/chapter10_tests.R]: Thank you for pointing that out, Professor. I have removed the extraneous blank lines.

Comment (david-yz-liu) [reply exercises/chapter 10/chapter10_tests.R]: @mani1381 No it's not necessary to add additional blank lines, that's just a quirk of how GitHub renders files.

**Review by david-yz-liu — APPROVED** [review]

**Pull request merged**